homeassistant/components/sensor/tof.py

```diff
@@ -0,0 +1,115 @@
+"""
+Platform for tof
+Time of Flight - VL53L1X Laser Ranger.
+
+For more details about this platform, please refer to
+https://github.com/josemotta/vl53l1x-python
+
+Fixed setup for current driver version:
+
+- DEFAULT_RANGE is always LONG
+- A GPIO connected to VL53L1X XSHUT input resets the device.
+- XSHUT starts pulsing LOW and after that it is kept HIGH all time.
+
+"""
+from datetime import timedelta
+from functools import partial
+import logging
+import time
+import voluptuous as vol
+
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.components import rpi_gpio
+import homeassistant.helpers.config_validation as cv
+from homeassistant.const import CONF_NAME
+from homeassistant.helpers.entity import Entity
+
+REQUIREMENTS = ['smbus2==0.2.2', 'VL53L1X2==0.1.5']
+
+DEPENDENCIES = ['rpi_gpio']
+
+_LOGGER = logging.getLogger(__name__)
+
+LENGTH_MILIMETERS = 'mm'
+
+CONF_I2C_ADDRESS = 'i2c_address'
+CONF_I2C_BUS = 'i2c_bus'
+CONF_RANGE = 'range'
+CONF_XSHUT = 'xshut'
+
+DEFAULT_NAME = 'VL53L1X'
+DEFAULT_I2C_ADDRESS = 0x29
+DEFAULT_XSHUT = 16
+DEFAULT_SENSOR_ID = 123
+
+MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=5)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_NAME,
+                 default=DEFAULT_NAME): cv.string,
+    vol.Optional(CONF_I2C_ADDRESS,
+                 default=DEFAULT_I2C_ADDRESS): vol.Coerce(int),
+    vol.Optional(CONF_XSHUT,
+                 default=DEFAULT_XSHUT): cv.positive_int,
+})
+
+
+async def async_setup_platform(hass,
+                               config,
+                               async_add_entities,
+                               discovery_info=None):
+    """Setup the VL53L1X ToF Sensor from ST."""
+    from VL53L1X2 import VL53L1X  # pylint: disable=import-error
+
+    name = config.get(CONF_NAME)
+    unit = LENGTH_MILIMETERS
+    xshut = config.get(CONF_XSHUT)
+
+    #  pulse XSHUT port and keep it HIGH
+    rpi_gpio.setup_output(xshut)
+    rpi_gpio.write_output(xshut, 0)
+    time.sleep(0.01)
+    rpi_gpio.write_output(xshut, 1)
+    time.sleep(0.01)
+
+    sensor = await hass.async_add_job(partial(VL53L1X))
```

Comment:
`hass.async_add_job` is legacy. Please use `hass.async_add_executor_job` to schedule the call in the executor thread pool.

Reply:
sensor = await hass.async_add_executor_job(partial(VL53L1X))

Reply:
Is it ok this way?

Reply:
If `VL53L1X` is a callable, I don't think we need `partial`.

Reply:
Ops, I tried:
    sensor = await hass.async_add_executor_job(
        VL53L1X(bus_number)
    )
And I got this error: TypeError: 'VL53L1X' object is not callable

This seems to be fine:
    sensor = await hass.async_add_executor_job(
        partial(VL53L1X, bus_number)
    )